openhexa/sdk/pipelines/runtime.py

```diff
@@ -36,8 +36,12 @@ class PipelineParameterSpecs:
 
     def __post_init__(self):
         """Validate the parameter and set default values."""
-        if self.default and self.choices and self.default not in self.choices:
-            raise ValueError(f"Default value '{self.default}' not in choices {self.choices}")
+        if self.default and self.choices:
```

Comment:
So you are replicating the check you have in the `Parameter`. I think we should construct Parameter and Pipeline objects with the AST to properly validate all the fields as in those classes

Reply:
As of my understanding this is what we do, we construct pipeline from ast, by going through the tree and validating the parameters. 
But let's discuss it tomorrow